libyul/YulStack.cpp

```diff
@@ -319,10 +319,49 @@ Json::Value YulStack::astJson() const
 	return  m_parserResult->toJson();
 }
 
+Json::Value YulStack::assemblyJson() const
+{
+	yulAssert(assembleEVMWithDeployed().first);
+	std::shared_ptr<evmasm::Assembly> assembly{assembleEVMWithDeployed().first};
```

Comment:
`assembleEVMWithDeployed()` is a **very** heavy operation. It runs the whole Yul Optimizer. We definitely do not want to run it twice. Please reuse the result.

```suggestion
	std::shared_ptr<evmasm::Assembly> assembly = assembleEVMWithDeployed().first;
	yulAssert(assembly);
```

Reply:
Also, note that `YulStack::assemble()`, which we run in `CommandLineInterface` already runs that function, so even a single call is redundant. We should just reuse the assembly we produce there.

I actually wonder if the assemblies are even guaranteed to be produced deterministically from the source if we don't reset the stack in the meantime. There could be some global state there.

Reply:
Ok, I just removed `YulStack::assemblyJson()`  now. I just added an `assemblyJson` member in `MachineAssemblyObject`, where this is now just constructed within `assembleWithDeployed`.

Reply:
It's much better now, but I'd simplify it even further: https://github.com/ethereum/solidity/pull/14612#discussion_r1603608333

Basically, we don't even need to store it. We could just have a helper that computes it from `MachineAssemblyObject`.